jaxrs-spec/src/main/asciidoc/chapters/environment/_javaee.adoc

```diff
@@ -197,8 +197,12 @@ The following additional requirements apply when using Managed Beans,
 CDI-style Beans or EJBs as resource classes, providers or `Application`
 subclasses:
 
-* Field and property injection of JAX-RS resources MUST be performed
-prior to the container invoking any `@PostConstruct` annotated method.
+* For JAX-RS resources where the JAX-RS implementation participates in
+their creation and initialization, field and property injection in these
+resources MUST be performed prior to the container invoking any 
+`@PostConstruct` annotated method. For resources created by the
+application (i.e. instances returned via the `Application.getSingletons()`
```

Comment:
I would replace `i.e.` with `e.g.`
The same happens when `configuration.register` is utilized. The customers are surprised that the fields were not injected when they provide instances.

Reply:
Or maybe it makes more sense to clarify whole `The following additional requirements apply when using Managed Beans,
CDI-style Beans or EJBs as resource classes, providers or `Application`
subclasses` at line 196?

Reply:
@jansupol Between the two suggestions, I prefer your first one ("i.e." to "e.g.", and mentioning the `configuration.register` approach too) more because the other bullet points in this section should still be applicable whether JAX-RS is involved in the bean creation or not.  I'll code that up shortly.  Thanks for the feedback!

Reply:
Wording changes are in.  @jansupol and @spericas, can you take another look? Thanks!